### ISSUE TITLE
MariaDB-10.6 added syntax for SELECT ... OFFSET ... FETCH [WITH TIES]

### ISSUE THREAD

**Issue by grooverdan** [sqlalchemy/sqlalchemy#6999]
### Describe the use case

Like Oracle, PostgreSQL, and MSSQL; MariaDB has implemented fetching with WITH TIES

This is used for fetching multiple results, and if the rows are identical at the end of the LIMIT, multiple rows are returned (identical to other DBs).

https://docs.sqlalchemy.org/en/14/core/selectable.html?highlight=with_ties#sqlalchemy.sql.expression.GenerativeSelect.fetch

### Databases / Backends / Drivers targeted

mariadb

### Example Use

```select(table_a.c.a).order_by(table_a.c.a).fetch(2, with_ties=True)```

Like existing implementations.

### Additional context

MariaDB reference: https://mariadb.com/kb/en/select-offset-fetch/

**Comment by CaselIT** [comment]
Thanks for the report.

If the syntax is standard it should be easy to add

**Comment by grooverdan** [comment]
Yep, follows same as Postgres and SQL-2011 ( https://jakewheat.github.io/sql-overview/sql-2011-foundation-grammar.html#fetch-first-clause) but with _fetch first percentage_ isn't implemented,  `OFFSET` and `FETCH` have simple values like `LIMIT`.

**Comment by CaselIT** [comment]
@grooverdan just to confirm, are you working on this?

**Comment by grooverdan** [comment]
not currently. I might eventually, just got a few other things to finish off too.

**Comment by CaselIT** [comment]
ok, I'll try working on it if I get the change

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **master** branch:

**Enable tests of fetch/offset for mariadb>=10.6** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3128

**Comment by grooverdan** [comment]
> It's actually already supported, we just need to update the test requirements.

:-)

I looked though the reviewboard. Change looks nice and simple. Thanks for writing it.

> it this expected or we are doing something strange?

Its expected. Its https://jakewheat.github.io/sql-overview/sql-2011-foundation-grammar.html#offset-row-count -> simple value specification that gets to be a literal and not an expression.

But given postgres, mssql, and oracle support it, we may do this in the future.

**Comment by grooverdan** [comment]
Thanks @CaselIT
